src/aws-cpp-sdk-core/include/smithy/client/serializer/JsonOutcomeSerializer.h

```diff
@@ -0,0 +1,99 @@
+/**
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#pragma once
+#include <aws/core/AmazonWebServiceResult.h>
+#include <aws/core/client/AWSError.h>
+#include <aws/core/utils/Outcome.h>
+#include <aws/core/utils/json/JsonSerializer.h>
+#include <smithy/tracing/TracingUtils.h>
+#include <smithy/tracing/TelemetryProvider.h>
+
+namespace smithy
+{
+    namespace client
+    {
+        using TracingUtils = components::tracing::TracingUtils;
+        using CoreErrors = Aws::Client::CoreErrors;
+        using AWSError = Aws::Client::AWSError<CoreErrors>;
+        using JsonValue = Aws::Utils::Json::JsonValue;
+        using HttpResponseOutcome = Aws::Utils::Outcome<std::shared_ptr<Aws::Http::HttpResponse>, AWSError>;
+        using JsonOutcome = Aws::Utils::Outcome<Aws::AmazonWebServiceResult<JsonValue>, AWSError>;
+        using TelemetryProvider = components::tracing::TelemetryProvider;
+
+        class JsonOutcomeSerializer
+        {
+        public:
+            explicit JsonOutcomeSerializer(const std::shared_ptr<TelemetryProvider> telemetryProvider)
```

Comment:
you'd prefer pass-by-value-and-move or pass-by-const-reference-and-copy

passing by const value will create an unnecessary temporary copy of telemetryProvider shared_ptr (which is relatively somewhat expensive).

Reply:
yeah youre right i think i missed this somewhere in doing the change, good eye